components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -560,6 +569,63 @@ public static boolean authenticateClient(String clientId, String clientSecretPro
         return true;
     }
 
+    /**
+     * Authenticate the OAuth Consumer.
+     *
+     * @param clientId             Consumer Key/ Id.
+     * @param clientSecretProvided Consumer Secret issued during the time of registration.
+     * @param appTenant            Tenant domain of the application.
+     * @return true, if the authentication is successful, false otherwise.
+     * @throws IdentityOAuthAdminException Error when looking up the credentials from the database
+     */
+    public static boolean authenticateClient(String clientId, String clientSecretProvided, String appTenant)
+            throws IdentityOAuthAdminException, IdentityOAuth2Exception, InvalidOAuthClientException {
+
+        OAuthAppDO appDO = OAuth2Util.getAppInformationByClientId(clientId, appTenant);
+        if (appDO == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Cannot find a valid application with the provided client_id: " + clientId);
```

Comment:
If this is not logged as an error in ay other place, please convert this to a error log.